lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -136,91 +150,146 @@ def __init__(
             tls_directory_path: Path of the directory
                 where certificates should be stored on the workload.
             workload: Either a Container or a Machine.
+            common_name: The common name of the certificate
+            sans_dns: Subject alternative names of the certificate
+            sans_ip: Subject alternative IP addresses of the certificate
         """
         super().__init__(charm, "tls")
         self.charm = charm
         self.workload = workload
         self._service_name = service_name
         self.tls_directory_path = tls_directory_path
-        self.tls_access = TLSCertificatesRequiresV3(charm, TLS_CERTIFICATE_ACCESS_RELATION_NAME)
+        self.common_name = common_name
+        self.sans_dns = sans_dns
+        self.sans_ip = sans_ip
+        self.mode = self._get_mode()
```

Comment:
[minor] It feels a little odd to set this at the class level when it's only used immediately after.

I wonder if using a property, or even a local variable makes more sense.

This implementation is a lot easier to follow, though, so I like that.